File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start=0;
        int end=height.length-1;
        int max_water=Integer.MIN_VALUE;
        int ans=0;
        while(start<end){
            ans=Math.min(height[start],height[end])*(end-start);
            max_water=Math.max(max_water,ans);
            if(height[start]<height[end]){
                start++;
            }
            else if(height[start]>height[end]){
                end--;
            }
            else{
                start++;
                end--;
            }
            
        }
        return max_water;
        
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start=0,end=height.size()-1;
        int ans=INT_MIN;
        while(start<end){
            int temp=min(height[start],height[end])*(end-start);
            ans=max(ans,temp);
            if(height[start]<height[end]){
                start++;
            }
            else if(height[start]>height[end]){
                end--;
            }
            else{
                start++;
                end--;
            }
        }
        return ans;
        
    }
};